### PR TITLE
Fix silent Qwen3.5/3.6 corruption from double RMSNorm shift

### PR DESCRIPTION
**Pull request by avlp12**
Fixes the root cause behind #1197. This overlaps with #1623, which reaches the same conclusion independently — I am opening this as an alternative in case that one has stalled, and because I have end-to-end evidence on a different architecture plus a regression case it does not cover. Happy to close this in favour of #1623 if a maintainer prefers it.

## The bug

`Qwen3_5` `sanitize` treats the presence of MTP tensors as evidence that a checkpoint is still in raw HF form:

```python
should_shift_norm_weights = has_mtp_weights or has_unsanitized_conv1d
```

That is not what MTP presence means. A checkpoint can be fully converted — Conv1D already in MLX layout, RMSNorm weights already shifted — and still ship its MTP head. Loading it shifts every norm a second time (γ 0.94 → 1.94). No error is raised anywhere; the model just generates noise. That is the failure mode reported in #1197, and it is easy to misattribute to the quantization.

The Conv1D layout already carries the signal reliably, and it is what the conversion itself keys off, so I use it alone.

## Reproduction

I hit this on `Qwen/Qwen3.8-27B` (dense `qwen3_5`, 64 layers, hybrid GatedDeltaNet/full-attention), converted to 4-bit with the MTP head preserved. Teacher-forced NLL on a fixed Korean passage, identical weights, identical prompt:

| mlx-lm | NLL |
| --- | --- |
| current `main` | 17.460 |
| this patch | 1.679 |

17.46 is worse than uniform over a 248k vocab, so the logits are actively wrong rather than merely degraded. The number reproduces to three decimals across runs, and generation through `mlx_lm.generate` is fluent once patched. Builds without an MTP head were never affected, which is why this hides so well — it looks like the MTP-bearing quantization is what broke.

This is a different model from the one in #1197 (dense 27B here, 35B MoE there), so the two reports together cover both branches of the family.

## What changed

- drop MTP presence from the predicate; keep the raw-Conv1D check
- extend the existing regression test with the case that reaches users: an already-converted checkpoint that *still contains* MTP tensors must load unshifted

The existing test only re-sanitized weights that had already had their MTP keys stripped, so it could not catch this. Reverting the one-line predicate change makes the new assertion fail.

## Validation

- `python -m unittest tests.test_models`: 78 passed, 1 skipped
- `black --check mlx_lm/models/qwen3_5.py tests/test_models.py`: clean
- end-to-end on the 27B checkpoint above, before and after

### CODE REVIEW

**Comment by AirRunner** [comment]
Confirming this is fixed independently in PR #990 too, same conclusion: dropping MTP presence from `should_shift_norm_weights` and keying only off the raw Conv1D layout.

**Comment by zcbenz** [comment]
Closing in favor of #1623.
